src/lib.rs

```diff
@@ -45,15 +45,57 @@ pub struct Header {
     pub generator: u32,
 }
 
+/// Forces early fragment tests. See https://www.khronos.org/opengl/wiki/Early_Fragment_Test.
+/// In GLSL: layout(early_fragment_tests) in;
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct EarlyFragmentTest {}
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum FragmentDepthWrite {
+    ///
+    Greater,
+
+    ///
+    Less,
+
+    ///
+    Unchanged,
+}
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum PixelCenter {
+    /// Pixels appear centered on whole-number pixel offsets. E.g., the coordinate (0.5, 0.5) appears to move to (0.0, 0.0).
+    Integer,
+
+    /// Pixels appear centered at offsets of (0.5, 0.5) from whole numbers
+    Offset,
+}
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum FragmentCoordinateOrigin {
+    ///
+    UpperLeft,
+
+    ///
+    LowerLeft,
+}
+
 /// Stage of the programmable pipeline.
 #[derive(Clone, Copy, Debug, PartialEq)]
 #[cfg_attr(feature = "serialize", derive(Serialize))]
 #[cfg_attr(feature = "deserialize", derive(Deserialize))]
 #[allow(missing_docs)] // The names are self evident
 pub enum ShaderStage {
     Vertex,
-    Fragment,
-    Compute,
+    Fragment {
+        pixel_center: PixelCenter,
+        coordinate_origin: FragmentCoordinateOrigin,
+        early_fragment_tests: Option<EarlyFragmentTest>,
+        depth_write: Option<FragmentDepthWrite>,
```

Comment:
actually, since it's an option, it feels like `Any`/`Unchanged` are redundant?

Reply:
Any == None
Unchanged == Some(Unchaged)

Confusing, I know :D.